plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -120,6 +120,65 @@ let mk_cmd_pattern value =
   let name = String.capitalize_ascii (str_of_ident value.id) |> lident in
   ppat_construct name args
 
+let ty_show_of_core_type inst ty =
+  let rec aux ty =
+    let open Ppxlib in
+    match ty.ptyp_desc with
+    | Ptyp_var v -> (
+        match List.assoc_opt v inst with None -> evar "char" | Some t -> aux t)
+    | Ptyp_constr (id, []) -> pexp_ident id
+    | Ptyp_constr (id, tys) ->
+        let args =
+          List.map
+            (fun t ->
+              let e = aux t in
+              (Nolabel, e))
+            tys
+        in
+        pexp_apply (pexp_ident id) args
+    | _ ->
+        failwith "shouldn't happen (unsupported types should be caught before)"
+  in
+  aux ty
+
+let exp_of_ident id = pexp_ident (lident (str_of_ident id))
+
+let call config esut value =
```

Comment:
Will that function be used elsewhere than in `run_case`? I’d propose to move it inside `run_case` and/or give it a name that’s easier to decode without context (`function_call_exp`?).